From 98b1f938f7d2399ea4492b130c7e5b95c937db9b Mon Sep 17 00:00:00 2001
Message-Id: <patch-1.1-98b1f938f7d-20210417T122959Z-avarab@gmail.com>
In-Reply-To: <patch-1.1-83266f30b67-20210417T084346Z-avarab@gmail.com>
References: <patch-1.1-83266f30b67-20210417T084346Z-avarab@gmail.com>
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?=
 <avarab@gmail.com>
Date: Sat, 17 Apr 2021 10:37:17 +0200
Subject: [PATCH v2] CodingGuidelines: remove suggestion to write commands in
 Perl/SH
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Remove a suggestion to write new commands in Perl or Shell to
experiment. This advice was added in 6d0618a820a (Add
Documentation/CodingGuidelines, 2007-11-08).

Since then the consensus changed to having no new such commands unless
necessary, and existing ones have been actively migrated to C.

So this isn't a new proposal or a suggestion to change the coding
style, but bringing this stale part of the CodingGuidelines in line
with reality.

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---

V1 of this was misencoded. I sent it as UTF-8 with
charset=ISO-2022-JP.

I'd been hacking the test-lib.sh, and at one point it escaped its
quarantine and started setting config in my main git.git, and I'd
managed to miss that i18n.logOutputEncoding encoding variable when cleaning it up.

Range-diff against v1:
1:  83266f30b67 ! 1:  98b1f938f7d CodingGuidelines: remove suggestion to write commands in Perl/SH
    @@ Commit message
         Since then the consensus changed to having no new such commands unless
         necessary, and existing ones have been actively migrated to C.
     
    +    So this isn't a new proposal or a suggestion to change the coding
    +    style, but bringing this stale part of the CodingGuidelines in line
    +    with reality.
    +
         Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
     
      ## Documentation/CodingGuidelines ##

 Documentation/CodingGuidelines | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/Documentation/CodingGuidelines b/Documentation/CodingGuidelines
index 45465bc0c98..b9cd55db6a8 100644
--- a/Documentation/CodingGuidelines
+++ b/Documentation/CodingGuidelines
@@ -416,11 +416,6 @@ For C programs:
    that are made available to it by including one of the header files
    it must include by the previous rule.
 
- - If you are planning a new command, consider writing it in shell
-   or perl first, so that changes in semantics can be easily
-   changed and discussed.  Many Git commands started out like
-   that, and a few are still scripts.
-
  - Avoid introducing a new dependency into Git. This means you
    usually should stay away from scripting languages not already
    used in the Git core command set (unless your command is clearly
-- 
2.31.1.722.g788886f50a2

